backends/tests/test_sycl_queue_manager.cpp

```diff
@@ -123,19 +170,33 @@ TEST_F (TestDPPLSyclQueueManager, CheckGetNumActivatedQueues)
     DPPLQueue_Delete(q);
 }
 
-
 TEST_F (TestDPPLSyclQueueManager, CheckDPPLDumpDeviceInfo)
 {
     auto q = DPPLQueueMgr_GetCurrentQueue();
     EXPECT_NO_FATAL_FAILURE(DPPLDevice_DumpInfo(DPPLQueue_GetDevice(q)));
     EXPECT_NO_FATAL_FAILURE(DPPLQueue_Delete(q));
 }
 
+TEST_F (TestDPPLSyclQueueManager, CheckIsCurrentQueue)
+{
+    if(!DPPLQueueMgr_GetNumQueues(DPPL_OPENCL, DPPL_GPU))
+        GTEST_SKIP_("No OpenCL GPU.\n");
+
+    auto Q0 = DPPLQueueMgr_GetCurrentQueue();
+    EXPECT_TRUE(DPPLQueueMgr_IsCurrentQueue(Q0));
+    auto Q = DPPLQueueMgr_PushQueue(DPPL_OPENCL, DPPL_GPU, 0);
+    EXPECT_TRUE(DPPLQueueMgr_IsCurrentQueue(Q));
+    EXPECT_FALSE(DPPLQueueMgr_IsCurrentQueue(Q0));
+    DPPLQueue_Delete(Q);
+    DPPLQueueMgr_PopQueue();
+    EXPECT_TRUE(DPPLQueueMgr_IsCurrentQueue(Q0));
+    DPPLQueue_Delete(Q0);
+}
 
 int
 main (int argc, char** argv)
 {
-  ::testing::InitGoogleTest(&argc, argv);
-  int ret = RUN_ALL_TESTS();
-  return ret;
+    ::testing::InitGoogleTest(&argc, argv);
+    int ret = RUN_ALL_TESTS();
+    return ret;
```

Comment:
It is possible to link each test case executable with gmain lib and do not create main() function in each cpp file. Lets do it in separate PR.

Reply:
Opened a new issues #99